### PR TITLE
Add option to log fine-tune checkpoints to wandb

### PR DESCRIPTION
**Pull request by morganmcg1**
This PR:

- Adds optional logging of checkpoints to Weights & Biases, with the same frequency as the google drive frequency
- Adds an open-in-colab badge to the Finetuning notebook
- Fixes an PL warning to use `devices=n_gpus` instead of `gpus=n_gpus`

Tested:
- save_models_to_google_drive = True | save_wandb = True
- save_models_to_google_drive = False | save_wandb = True
- - save_models_to_google_drive = False | save_wandb = False

**Try it yourself**
[Colab using the `add_wandb_artifacts_checkpointing` branch](https://colab.research.google.com/drive/1NgKCbw8xE1P-7e_4Y1x55Ukgv59auriN?usp=sharing)

**Examples**

- Example wandb run: https://wandb.ai/wandb_gen/dd-drums-finetune/runs/98yvisk4
- [Weights & Biases Artifacts page](https://wandb.ai/wandb_gen/dd-drums-finetune/artifacts) from above project with the checkpoint
- Screenshot of checkpoint saved in Weights & Biases alongside all the training  metadata at that point:

Screenshot 2022-10-05 at 16 15 01

- Screenshot of the optional `wandb` selector:
Screenshot 2022-10-05 at 16 23 09

### CODE REVIEW

**Comment by twobob** [comment]
Hi. might I ask what is the result when storage is completely full. does it just refuse further files or does it cease all logging of data.  Thanks.

**Comment by morganmcg1** [comment]
@twobob right now there is no programmatic enforcement when you hit the 100GB free limit, you will probably just start to get some emails from sales. If we do enforce programmatically, we'll make sure that metric logging isn't impacted

**Comment by zqevans** [comment]
Could you test this with save_models_to_wandb set to False? I'm pretty sure it'll log it anyways with this setup. Boolean CLI args are weird and it ends up just testing for existence. So even if you pass in "False", it just sets it to True.

**Comment by morganmcg1** [comment]
Hey @zqevans , good point on the weird behaviour around passing arguments, this fix has been sitting on my machine for a week, finally getting around to pushing it here. I also added a link to a colab the runs from my branch here in case you'd like to test for yourself
